File: result/translate/transcoder/python2java/java/CHECK_STRING_FOLLOWS_ANBN_PATTERN_NOT.java
public static boolean f_gold ( String str ) {
  int n = str . length ( ) ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    if ( ( str . charAt ( i ) != 'a' ) && ( str . charAt ( i ) != 'b' ) ) {
      break ;
    }
  }
  if ( ( i * 2 ) != n ) {
    return false ;
  }
  for ( int j = i ;
  j < n ;
  j ++ ) {
    if ( ( str . charAt ( j ) != 'b' ) && ( str . charAt ( j ) != 'a' ) ) {
      return false ;
    }
  }
  return true ;
}

File: result/translate/starchat/java2cpp/c++/CHECK_STRING_FOLLOWS_ANBN_PATTERN_NOT.cpp
bool f_gold ( string s ) {
        int l = s.length();
        if (l % 2 == 1) return false;
        int i = 0, j = l - 1;
        while (i < j) {
            if (s[i]!= 'a' || s[j]!= 'b') return false;
            i++, j--;
        }
        return true;
    }